src/client/app/actions/barReadings.ts

```diff
@@ -44,7 +44,7 @@ export function shouldFetchMeterBarReadings(state: State, meterID: number, timeI
  * @returns {boolean} True if the readings for the given group, time, and duration are missing; false otherwise.
  */
 export function shouldFetchGroupBarReadings(state: State, groupID: number, timeInterval: TimeInterval, barDuration: moment.Duration): boolean {
-	const readingsForID = state.readings.bar.byMeterID[groupID];
+	const readingsForID = state.readings.bar.byGroupID[groupID];
```

Comment:
I think this is an okay change but it does not seem to relate to the map group changes. (Same thought for compareReadings.ts) Is it correct that this fixes a mistake that has been in the code for a while? I did some quick testing (not extensive) and it may be the case that issue #579 is fixed by this change. Would you be willing to see if it seems gone for you and link that issue if it is the case? (Also add comment to issue that think okay but it has had to find so closing for now.)

Reply:
issue #579 is fixed supposedly, if my recreation of the bug is right. I will add it into this pull request as well.